src/keys/bip39.rs

```diff
@@ -19,7 +19,23 @@ use bitcoin::Network;
 
 use miniscript::ScriptContext;
 
-pub use bip39::{Language, Mnemonic, MnemonicType, Seed};
+pub use bip39::{Language, Mnemonic};
+
+type Seed = [u8; 64];
+
+/// Type describing entropy length (aka word count) in the mnemonic
+pub enum WordCount {
```

Comment:
Is there a reason we don't want to name this `MnemonicType`?

Reply:
I ask because it kind of used to exist earlier: https://docs.rs/bip39/0.5.1/bip39/enum.MnemonicType.html

Reply:
Right point. Reason is, `WordCount` seems more intuitive to me than `MnemonicType`. I had to look it up to see what it was doing. 

`MnemonicType` is an enum used in `tiny-bip39` which does more than simple word or entropy counting. It makes sense for the context of `tiny-bip39` impls, but it seemed too much for `rust-bip39` to use directly. It just takes an `usize` value for it.   

So, I decided to make our own simple enum for it and it made sense to give it an easier name.

Reply:
PS: I think it should be `WordsCount` instead of `WordCount`.

Reply:
I agree with @artfuldev that `MnemonicType` is less of a code change for downstream users but if you want to change it to something more meaningful I like your original name `WordCount` better.  If we're talking about how many words are in an article or book or something we generally say "the word count is 20" .. not "the words count is 20". See also: https://trends.google.com/trends/explore?q=%22words%20count%22,%22word%20count%22

Reply:
oops, it's already merged. `WordsCount` will be fine. 😉

Reply:
Ah, thats a nice google trick to sort this kinda disputes.. 😂 

My bad, I will sneak the change back in some future PR.

Reply:
BTW that reminds me that this is probably a downstream API change. Even if we kept the name `MnemonicType` the bip39 API still changes. Maybe we should better document the API change somewhere?

Reply:
Since it does change the API we probably should mention it in the changelog. We can add that in another PR or when we make the next release branch we can add it then.

Reply:
Updated in https://github.com/bitcoindevkit/bdk/pull/464